pkg/filter/filter.go

```diff
@@ -180,7 +195,27 @@ func (f *Filter) shouldDiscardByBuiltInDDLAllowlist(ddlType model.ActionType) bo
 	return true
 }
 
-// IsSysSchema returns true if the given schema is a system schema
-func IsSysSchema(db string) bool {
+// isSysSchema returns true if the given schema is a system schema.
+func isSysSchema(db string) bool {
 	return filterV1.IsSystemSchema(db)
 }
+
+// isUserRelatedSysTable returns true if the given table is a system schema
+// related to user permissions.
+// Currently we are not considering the problem of different
+// user table fields due to inconsistent upstream and downstream TiDB versions.
+// Also we did not consider the cooperation with other filtering rules and
+// will copy all the permissions directly.
+// See also: https://github.com/pingcap/ticdc/issues/2192#issuecomment-872096744.
+// NOTICE: If you filter the user-related tables in the filter rule it will still take effect.
+func isUserRelatedSysTable(db, table string) bool {
+	// First we need to check if it is a MYSQL system table.
+	// See also: https://dev.mysql.com/doc/refman/8.0/en/system-schema.html.
+	if strings.ToUpper(db) == "MYSQL" {
+		if _, ok := userRelatedTables[table]; ok {
```

Comment:
Do we need `strings.ToLower(table)`?

Reply:
Thanks for your review!
I don't think it's necessary, because in general these tables should always be lowercase. I'm using upper case here to align with the logic in our filter package.